File: Clojure/Hanoi/hanoi.clj
(defn hanoi [n from to via]
  (if (> n 1)
    (do
      (hanoi (dec n) from via to)
      (println (format "%s -> %s" from to))
      (hanoi (dec n) via to from))
    (println (format "%s -> %s" from to))))

(defn main []
  (let [n (read-line)]
    (hanoi (read-string n) 'a 'b 'c)))

(main)

File: Go/Hanoi/hanoi.go
package main

import "fmt"

func hanoi(n int, from, to, via string) {
  if n > 1 {
    hanoi(n - 1, from, via, to)
    fmt.Println(from, " -> ", to)
    hanoi(n - 1, via, to, from)
  } else {
    fmt.Println(from, " -> ", to)
  }
}

func main() {
  var n int
  fmt.Scan(&n)
  hanoi(n, "a", "b", "c")
}
